src/amo/utils/addonVersions.ts

```diff
@@ -4,25 +4,72 @@ import * as vscode from "vscode";
 import constants from "../../config/config";
 import { addonVersion } from "../types";
 
-export async function getAddonVersions(input: string, next?: string) {
-  if (next) {
-    const url = next;
-    const response = await fetch(url);
-    const json = await response.json();
-    return json;
+async function getPaginatedVersions(input: string, next: string) {
+  const url = next;
+  const response = await fetch(url);
+  if (!response.ok) {
+    await vscode.window
+      .showErrorMessage(
+        `(Status ${response.status}): Could not fetch versions.`,
+        { modal: true },
+        { title: "Try Again" },
+        { title: "Fetch New Addon" }
+      )
+      .then(async (action) => {
+        if (action?.title === "Try Again") {
+          return await getAddonVersions(input, next);
+        } else if (action?.title === "Fetch New Addon") {
+          vscode.commands.executeCommand("assay.get");
+          throw new Error("Process restarted");
+        } else {
+          throw new Error("Failed to fetch versions");
```

Comment:
same comments about using `const` and refactoring out some shared code here, and in `getFirstVersions`